packages/ebsi-authorization-client/src/agent/EBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,363 @@
+  }
+  private configuration: GetOIDProviderMetadataSuccessResponse
+
+  //TODO Pass the configuration manually or call init
+  constructor(readonly baseUrl: string = 'https://api-pilot.ebsi.eu/authorisation/v4/') {
+    // TODO call configuration endpoint
+  }
+
+  async authorize(
+    args: { credential: CredentialPayload; definitionId: ScopeByDefinition; keyPair: { privateKeyHex: string; publicKeyHex: string } },
+    context: IRequiredContext,
+  ) {
+    const { credential, definitionId, keyPair } = args
+
+    const metadataResponse = await this.getOIDProviderMetadata()
+    if ('status' in metadataResponse) {
+      throw Error(JSON.stringify(metadataResponse))
+    }
+
+    const vc = await createVcJwt(
+      {
+        privateKeyHex: keyPair.privateKeyHex,
+        payloadVc: credential,
+      },
+      context,
+    )
+
+    const vpJwt = await createVPJwt(
+      {
+        keyPair,
+        did: this.configuration.issuer.toString(),
+        vc,
+        audience: this.configuration.issuer.toString(),
+        domain: this.baseUrl,
+      },
+      context,
+    )
+
+    let descriptorMap: Descriptor[] = []
+
+    if (definitionId === ScopeByDefinition.didr_invite_presentation) {
+      descriptorMap = [
+        {
+          id: 'didr_invite_credential',
+          format: 'jwt_vp',
+          path: '$',
+          path_nested: {
+            id: 'didr_invite_credential',
+            format: 'jwt_vc',
+            path: '$.vp.verifiableCredential[0]',
+          },
+        },
+      ]
+    } else if (definitionId === ScopeByDefinition.tir_invite_presentation) {
+      descriptorMap = [
+        {
+          id: 'tir_invite_credential',
+          format: 'jwt_vp',
+          path: '$',
+          path_nested: {
+            id: 'tir_invite_credential',
+            format: 'jwt_vc',
+            path: '$.vp.verifiableCredential[0]',
+          },
+        },
+      ]
+    } else if (definitionId === ScopeByDefinition.tnt_authorise_presentation) {
+      descriptorMap = [
+        {
+          id: 'tnt_authorise_credential',
+          format: 'jwt_vp',
+          path: '$',
+          path_nested: {
+            id: 'tnt_authorise_credential',
+            format: 'jwt_vc',
+            path: '$.vp.verifiableCredential[0]',
+          },
+        },
+      ]
+    }
+
+    const presentationSubmission = {
+      id: uuid(),
+      definition_id: definitionId,
+      descriptor_map: descriptorMap,
+    }
+
+    const tokenResponse = await this.getAccessToken({
+      grant_type: 'vp_token',
+      vp_token: vpJwt,
+      presentation_submission: presentationSubmission,
+      scope: OpenIDScope.didr_invite,
+    })
+    if ('status' in tokenResponse) {
+      throw new Error(JSON.stringify(tokenResponse))
+    }
+    return tokenResponse
+  }
+
+  private async getOIDProviderMetadata(): Promise<GetOIDProviderMetadataResponse> {
+    return await (
+      await fetch(`${this.baseUrl}/.well-known/openid-configuration`, {
```

Comment:
We should have generic fetch support, which does proper error handling. Either in ssi-types, sdk-core or express-support from the top of my head

Reply:
There only one of the 3 libraries that has cross-fetch as a dependency: ssi-sdk-core. It's used specifically to download images. What do I miss?